### PR TITLE
Derive `Serialize` for ZoneType

### PR DESCRIPTION
**Pull request by LEXUGE**
Add a few traits derivation for convenience.

### CODE REVIEW

**Comment by djc** [comment]
Hmm, not sure conditional Clone impls are a good idea.

**Comment by LEXUGE** [comment]
But I really need to clone FileAuthority to spawn it though.

**Comment by bluejekyll** [comment]
> But I really need to clone FileAuthority to spawn it though.

I think @djc 's concern is that it becomes part of the openssl/rustls evaluation. What we probably what instead is just an additional `#[derive(Clone)]` that would appear as a second line with no `cfg_attr` portion to it. So just add that instead of modifying the existing `cfg_attr` attribute.

File: crates/server/src/store/file/authority.rs [hunk 1]
```diff
@@ -30,6 +30,7 @@ use crate::store::in_memory::InMemoryAuthority;
 ///
 /// Authorities default to DNSClass IN. The ZoneType specifies if this should be treated as the
 /// start of authority for the zone, is a Secondary, or a cached zone.
+#[cfg_attr(not(any(feature = "openssl", feature = "dnssec-ring")), derive(Clone))]
```
Comment (bluejekyll) [inline crates/server/src/store/file/authority.rs:33]: just make this `#[derive(Clone)]` it's not clear why it should only be when there is no openssl/dnssec-ring support.

Comment (LEXUGE) [reply crates/server/src/store/file/authority.rs:33]: Cause Signer is not clone if ring or OpenSSL is enabled.

**Comment by LEXUGE** [comment]
Using arc solved my needs for clone. I would revise it to include only serialize.

File: crates/server/src/store/file/authority.rs (hunk 1 above)
Comment (bluejekyll) [reply crates/server/src/store/file/authority.rs:33]: This is probably an oversight. I think the OpenSSL types are cloneable, but the *ring* types are not in Signer. We can put the *ring* types behind an Arc to make that cloneable, which is what is recommended here:  https://github.com/briansmith/ring/issues/859#issuecomment-605129968

**Comment by LEXUGE** [comment]
@bluejekyll I have revised it.

**Review by bluejekyll — APPROVED** [review]
LGTM

**Pull request merged**